File: Java/Creational/Factory/Show/Code/src/TheRayCode/Factory/PetFactory.java
package TheRayCode.Factory;

public class PetFactory {
    public Pet createPet(String type) {
        if (type.equals("bird")) {
            return new Bird();
        } else if (type.equals("dog")) {
            return new Dog();
        } else if (type.equals("cat")) {
            return new Cat();
        } else {
            throw new IllegalArgumentException("Unknown pet type: " + type);
        }
    }
}

File: PHP/Creational/AbstractFactory/Show/PetFactory.php
<?php

namespace TheRayCode\Factory;

class PetFactory extends AbstractPetFactory
{
    public function createDog(): Dog
    {
        $species = $this->generateRandomSpecies();
        return new Dog($species);
    }

    public function createCat(): Cat
    {
        $species = $this->generateRandomSpecies();
        return new Cat($species);
    }

    private function generateRandomSpecies(): string
    {
        $speciesList = ['Labrador', 'Poodle', 'German Shepherd'];
        $randomIndex = array_rand($speciesList);
        return $speciesList[$randomIndex];
    }
}